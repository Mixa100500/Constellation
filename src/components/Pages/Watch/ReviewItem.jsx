import { Review } from "./styled"
import PropTypes from 'prop-types'

export const ReviewItem = ({ review }) => {
	return (
		<Review className='review'>
			<div className='review__icon-container'>
				<div className='review__icon-content'>
					<div className='review__icon'></div>
					<div className='review__icon-text'>{review.author[0]}</div>
				</div>
			</div>
			<div className='review__text'>
				<h3 className='review__author'>{review.author}</h3>
				<p className='review__content'>{review.content}</p>
				<p>Rating: {review.author_details.rating}</p>
			</div>
		</Review>
	)
}

ReviewItem.propTypes = {
  review: PropTypes.shape({
    author: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
    author_details: PropTypes.shape({
      rating: PropTypes.number
    })
  }).isRequired
}