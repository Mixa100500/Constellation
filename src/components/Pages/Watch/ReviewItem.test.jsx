import { render } from '@testing-library/react';
import { ReviewItem } from './ReviewItem';

describe('ReviewItem component', () => {
  const review = {
    author: 'John Doe',
    content: 'This is a review content.',
    author_details: {
      rating: 8
    }
  };

  test('renders author, content, and rating correctly', () => {
    const { getByText } = render(<ReviewItem review={review} />);
    
    expect(getByText('John Doe')).toBeInTheDocument();
    expect(getByText('This is a review content.')).toBeInTheDocument();
    expect(getByText('Rating: 8')).toBeInTheDocument();
  });

  test('displays first character of author name in icon text', () => {
    const { getByText } = render(<ReviewItem review={review} />);
    expect(getByText('J')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { container } = render(<ReviewItem review={review} />);
    expect(container).toMatchSnapshot();
  });

  test('renders with default rating if rating is not provided', () => {
    const reviewWithoutRating = {
      author: 'Jane Doe',
      content: 'This is another review content.',
      author_details: {}
    };
    const { getByText } = render(<ReviewItem review={reviewWithoutRating} />);
    
    expect(getByText('Rating:')).toBeInTheDocument();
  });
});