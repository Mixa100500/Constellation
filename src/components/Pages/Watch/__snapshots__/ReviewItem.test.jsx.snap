// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReviewItem component > matches snapshot 1`] = `
<div>
  <div
    class="sc-beySPh hJOgUm review"
  >
    <div
      class="review__icon-contrainer"
    >
      <div
        class="review__icon-content"
      >
        <div
          class="review__icon"
        />
        <div
          class="review__icon-text"
        >
          J
        </div>
      </div>
    </div>
    <div
      class="review__text"
    >
      <h3
        class="review__author"
      >
        John Doe
      </h3>
      <p
        class="review__content"
      >
        This is a review content.
      </p>
      <p>
        Rating: 
        8
      </p>
    </div>
  </div>
</div>
`;
